### ISSUE TITLE
Error loading shared library libgcc_s.so.1

### ISSUE THREAD

**Issue by Yevgenium** [lrybak/hassio-owserver#5]
After starting I get a lot of these errors:
```
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owserver)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
```

Full log:
```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Add-on: owserver (1-Wire)
 onewire server to read 1-Wire devices
-----------------------------------------------------------
 Add-on version: 0.2.1
 You are running the latest version of this add-on.
 System: Home Assistant OS 7.5  (armv7 / raspberrypi3)
 Home Assistant Core: 2022.3.8
 Home Assistant Supervisor: 2022.03.5
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] owserver.sh: executing... 
[23:05:31] INFO: Configuring fake device
[cont-init.d] owserver.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[23:05:32] INFO: Starting owserver...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owserver)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
[23:05:32] INFO: Starting owhttpd...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owhttpd)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
[23:05:32] INFO: Starting owserver...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owserver)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
[23:05:33] INFO: Starting owhttpd...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owhttpd)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
```

**Comment by lrybak** [comment]
I'm running it on x86_64, didn't test it on armv7. I will try to build test setup but first need to find the pi3 in my drawer :-)

**Comment by scubaandre** [comment]
Same results, new version of HASS OS:

-----------------------------------------------------------
 Add-on: owserver (1-Wire)
 onewire server to read 1-Wire devices
-----------------------------------------------------------
 Add-on version: 0.2.1
 You are running the latest version of this add-on.
 System: Home Assistant OS 7.6  (armv7 / raspberrypi3)
 Home Assistant Core: 2022.5.1
 Home Assistant Supervisor: 2022.05.0
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] owserver.sh: executing... 
[15:56:34] INFO: Configuring fake device
[cont-init.d] owserver.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[15:56:35] INFO: Starting owserver...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owserver)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
[15:56:35] INFO: Starting owhttpd...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owhttpd)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
[15:56:36] INFO: Starting owserver...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owserver)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
[15:56:36] INFO: Starting owhttpd...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owhttpd)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
[15:56:37] INFO: Starting owserver...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owserver)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
[15:56:37] INFO: Starting owhttpd...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owhttpd)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
[15:56:38] INFO: Starting owserver...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owserver)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
[15:56:38] INFO: Starting owhttpd...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owhttpd)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
[15:56:39] INFO: Starting owserver...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owserver)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
[15:56:39] INFO: Starting owhttpd...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owhttpd)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
[15:56:40] INFO: Starting owserver...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owserver)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
[15:56:40] INFO: Starting owhttpd...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owhttpd)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
[15:56:41] INFO: Starting owserver...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owserver)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
[15:56:41] INFO: Starting owhttpd...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owhttpd)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
[15:56:42] INFO: Starting owserver...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owserver)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
[15:56:42] INFO: Starting owhttpd...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owhttpd)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
[15:56:44] INFO: Starting owserver...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owserver)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
[15:56:44] INFO: Starting owhttpd...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owhttpd)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)

**Comment by Tomim7** [comment]
Same results

`[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Add-on: owserver (1-Wire)
 onewire server to read 1-Wire devices
-----------------------------------------------------------
 Add-on version: 0.2.1
 You are running the latest version of this add-on.
 System: Home Assistant OS 8.2  (armv7 / raspberrypi3)
 Home Assistant Core: 2022.7.5
 Home Assistant Supervisor: 2022.07.0
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] owserver.sh: executing... 
[23:38:15] INFO: Configuring fake device
[cont-init.d] owserver.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[23:38:15] INFO: Starting owserver...
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/bin/owserver)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /opt/owfs/lib/libow-3.2.so.4)
`

**Comment by alex88** [comment]
As a temporary workaround you can just exec into the container:
```
docker exec -ti addon_9e0ef242_owserver sh
```

and run

```
apk add --update --no-cache libgcc
```

**Comment by ChrisBela** [comment]
> As a temporary workaround you can just exec into the container:
> 
> ```
> docker exec -ti addon_9e0ef242_owserver sh
> ```
> 
> and run
> 
> ```
> apk add --update --no-cache libgcc
> ```

Just wondering if this issue will be corrected? I can issue the commands above to make it work but if my system restarts the library goes missing again and I have to do these commends again each time. Not sure why this change is not persistent, but I don't really know much about docker containers.

**Comment by lrybak** [comment]
I have identified and fixed the root cause. Fix is included in the [beta release 0.3.2-rc1](https://github.com/lrybak/addon-repository-beta/). I sincerely apologize for the delay in addressing this issue; it was due to time constraints on my end. Appreciate your patience. Please let me know if the fix is working for you. Once confirm I will merge and release stable version. Cheers

**Comment by lrybak** [comment]
Fixed in 0.3.2

**Comment by ChrisBela** [comment]
Thanks so much, been working great!!
